crates/biome_js_syntax/src/import_ext.rs

```diff
@@ -29,6 +32,36 @@ impl JsImport {
     }
 }
 
+impl AnyJsNamedImportSpecifier {
+    /// Type token of the import specifier.
+    ///
+    /// ```ts
+    /// import { type X }
+    ///          ^^^^
+    /// ```
+    pub fn type_token(&self) -> Option<JsSyntaxToken> {
+        match self {
+            Self::JsBogusNamedImportSpecifier(_) => None,
+            Self::JsNamedImportSpecifier(specifier) => specifier.type_token(),
+            Self::JsShorthandNamedImportSpecifier(specifier) => specifier.type_token(),
+        }
+    }
+
+    // Returns the import clause that includes this specifier.
+    pub fn import_named_clause(&self) -> Option<JsImportNamedClause> {
+        JsImportNamedClause::cast(self.syntax().ancestors().nth(3)?)
+    }
+
+    /// Returns `true` if this specifier or its import clause has a type modifier.
+    pub fn is_type_only(&self) -> bool {
```

Comment:
nit: `exports_only_types`

The name is misleading because it leads to "it's a type"